lib/internal/process/main_thread_only.js

```diff
@@ -154,6 +154,16 @@ function setupSignalHandlers(internalBinding) {
       delete signalWraps[type];
     }
   });
+
+  // re-arm pre-existing signal event registrations
+  // with this signal wrap capabilities.
+  const events = process.eventNames();
+  if (events != null && events !== undefined) {
+    events.forEach((ev) => {
+      if (isSignal(ev))
+        process.emit('newListener', ev);
+    });
+  }
```

Comment:
(might make sense to split this out into its own commit/PR)

Reply:
sure, will do

Reply:
moved this part to #24651